community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/InputIterator.java

```diff
@@ -103,6 +103,24 @@ public long position()
         {
             return actual.position();
         }
+
+        @Override
+        public int numberOfProcessors()
+        {
+            return actual.numberOfProcessors();
+        }
+
+        @Override
+        public boolean incrementNumberOfProcessors()
+        {
+            return actual.incrementNumberOfProcessors();
+        }
+
+        @Override
+        public boolean decrementNumberOfProcessors()
+        {
+            return actual.decrementNumberOfProcessors();
+        }
```

Comment:
You're absolutely right. I wonder... since there are many doing this delegation, could this be done differently, you think? Or perhaps those 4 methods could be reduced to one method:

```int changeNumberOfProcessors(int delta)```

where the returned value would be how many processors are assigned when the call exits.